### PR TITLE
Update temurin8 Cask for jdk8u422-b05.1 point release re-build

### PR DESCRIPTION
**Pull request by andrew-m-leonard**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [ ] `brew audit --cask --online <cask>` is error-free.
- [ ] `brew style --fix <cask>` reports no offenses.

---

### CODE REVIEW

**Comment by andrew-m-leonard** [comment]
Note, I have no idea what these two checkboxes mean?
brew audit --cask --online <cask> is error-free.
brew style --fix <cask> reports no offenses.

How does running those commands from my Mac relate to this PR?
I get:
```
anleonar@anleonar-mac homebrew-cask % brew audit --cask --online temurin@8
Fetching gem metadata from https://rubygems.org/.......
Fetching ast 2.4.2
Fetching parallel 1.25.1
Fetching language_server-protocol 3.17.0.3
Fetching json 2.7.2
Installing parallel 1.25.1
Installing ast 2.4.2
Fetching racc 1.8.0
Fetching rainbow 3.1.1
Installing language_server-protocol 3.17.0.3
Installing json 2.7.2 with native extensions
Installing racc 1.8.0 with native extensions
Installing rainbow 3.1.1
Fetching regexp_parser 2.9.2
Installing regexp_parser 2.9.2
Fetching ruby-progressbar 1.13.0
Fetching unicode-display_width 2.5.0
Installing ruby-progressbar 1.13.0
Installing unicode-display_width 2.5.0
Fetching parser 3.3.4.0
Installing parser 3.3.4.0
Fetching rubocop-ast 1.31.3
Installing rubocop-ast 1.31.3
Fetching rubocop 1.65.0
Installing rubocop 1.65.0
Fetching rubocop-md 1.2.2
Fetching rubocop-performance 1.21.1
Fetching rubocop-rspec 3.0.3
Fetching rubocop-sorbet 0.8.4
Installing rubocop-md 1.2.2
Installing rubocop-performance 1.21.1
Installing rubocop-rspec 3.0.3
Installing rubocop-sorbet 0.8.4
Bundle complete! 37 Gemfile dependencies, 28 gems now installed.
Bundled gems are installed into `../../../../opt/homebrew/Library/Homebrew/vendor/bundle`
Error: Cask 'temurin@8' is unavailable: No Cask with this name exists.
```

File: Casks/t/temurin@8.rb [hunk 1]
```diff
@@ -1,22 +1,29 @@
...
+  #   version string : remove ",1"
+  #   url : remove ".#{version.csv.fourth}"
+  #   regex : remove "\.(\d+)"
+  #   livecheck strategy version string : remove ",#{match[4]}"
 
-  url "https://github.com/adoptium/temurin8-binaries/releases/download/jdk#{version.csv.first}u#{version.csv.second}-b#{version.csv.third}/OpenJDK#{version.csv.first}U-jdk_x64_mac_hotspot_#{version.csv.first}u#{version.csv.second}b#{version.csv.third}.pkg",
+  version "8,422,05,1"
+  sha256 "809d7c0b6c7c2fd483711fb784e4fb101ad431c71dc8e96aa9c2cffaa8bfa7b7"
+
+  url "https://github.com/adoptium/temurin8-binaries/releases/download/jdk#{version.csv.first}u#{version.csv.second}-b#{version.csv.third}.#{version.csv.fourth}/OpenJDK#{version.csv.first}U-jdk_x64_mac_hotspot_#{version.csv.first}u#{version.csv.second}b#{version.csv.third}.pkg",
```
Comment (khipp) [inline Casks/t/temurin@8.rb:12]:
```suggestion
  version "8,422,05.1"
  sha256 "809d7c0b6c7c2fd483711fb784e4fb101ad431c71dc8e96aa9c2cffaa8bfa7b7"

  url "https://github.com/adoptium/temurin8-binaries/releases/download/jdk#{version.csv.first}u#{version.csv.second}-b#{version.csv.third}.#{version.csv.fourth}/OpenJDK#{version.csv.first}U-jdk_x64_mac_hotspot_#{version.csv.first}u#{version.csv.second}b#{version.csv.third.split(".").first}.pkg",
```

File: Casks/t/temurin@8.rb [hunk 2]
```diff
@@ -1,22 +1,29 @@
...
       verified: "github.com/adoptium/temurin8-binaries/"
   name "Eclipse Temurin 8"
   desc "JDK from the Eclipse Foundation (Adoptium)"
   homepage "https://adoptium.net/"
 
   livecheck do
     url "https://api.adoptium.net/v3/assets/feature_releases/8/ga?architecture=x64&image_type=jdk&jvm_impl=hotspot&os=mac&page=0&page_size=1&project=jdk&sort_method=DEFAULT&sort_order=DESC&vendor=eclipse"
-    regex(/^jdk(\d+)u(\d+)-b(\d+)$/i)
+    regex(/^jdk(\d+)u(\d+)-b(\d+)\.(\d+)$/i)
```
Comment (khipp) [inline Casks/t/temurin@8.rb:20]:
```suggestion
    regex(/^jdk(\d+)u(\d+)-b(\d+(?:\.\d+)*)$/i)
```

File: Casks/t/temurin@8.rb [hunk 3]
```diff
@@ -1,22 +1,29 @@
...
     strategy :json do |json, regex|
       json.map do |release|
         match = release["release_name"]&.match(regex)
         next if match.blank?
 
-        "#{match[1]},#{match[2]},#{match[3]}"
+        "#{match[1]},#{match[2]},#{match[3]},#{match[4]}"
```
Comment (khipp) [inline Casks/t/temurin@8.rb:26]:
```suggestion
        "#{match[1]},#{match[2]},#{match[3]}"
```

**Review by khipp — COMMENTED** [review]
Thank you for your contribution, @andrew-m-leonard!

To answer your questions, the cask token refers to its identifier, `temurin@8` (see [documentation](https://docs.brew.sh/Cask-Cookbook#header-line-details)).

The commands `brew audit --cask --online <cask>` and `brew style --fix <cask>` allow you to fix any outstanding issues before submitting the PR. `brew audit` looks for a corresponding file in your installed taps (refer to our [documentation on pull requests](https://docs.brew.sh/How-To-Open-a-Homebrew-Pull-Request#cask-related-pull-request) for more information).

As for the PR, I left a few suggestions below that would not require us to revert the changes on the next release.

**Comment by andrew-m-leonard** [comment]
> Thank you for your contribution, @andrew-m-leonard!
> 
> To answer your questions, the cask token refers to its identifier, `temurin@8` (see [documentation](https://docs.brew.sh/Cask-Cookbook#header-line-details)).
> 
> The commands `brew audit --cask --online <cask>` and `brew style --fix <cask>` allow you to fix any outstanding issues before submitting the PR. `brew audit` looks for a corresponding file in your installed taps (refer to our [documentation on pull requests](https://docs.brew.sh/How-To-Open-a-Homebrew-Pull-Request#cask-related-pull-request) for more information).
> 
> As for the PR, I left a few suggestions below that would not require us to revert the changes on the next release.

Thank you @khipp I have included your suggestions, which makes it future proof, thank you

**Review by p-linnane — APPROVED** [review]

**Pull request merged**